content/blog/cozycoder-vision/index.md

```diff
@@ -0,0 +1,69 @@
+
+But I am not very satisfied with the limited experience of chats. Let's take a simple example:
+You are setting up a new project - e.g. a microservice with an API in your enterprise environment. You start asking Copilot to add a HTTP server and an endpoint. You ask it for a database connection and some SQL to query the database. You get pretty far with it just by letting Copilot generate all the boilerplate code for you. Then you copy it to files and commit them. The next time, you wanna setup a new project, you start from scratch.
+
+But why? Why can't you just fork the previous chat at some point to reuse some parts of the conversation to create another project? Why can't the LLM just take the generated code and create a pull request in your companies Github enterprise installation? Why can't you share the conversation with your colleagues, so they can fork it and reuse it as well?
+
+## Giving Developers more Control
+
+The obvious problem with chat is that everyone is just treating it as chat between a human and a bot. But that's not true. Each time you interact with Copilot or other LLMs the full conversation is send to the LLM as a context. There is no reason, why this context couldn't be treated like a Git history instead.
+
+Let's imagine a tool that gives developers more control and go through the above example:
+You start a conversation with a LLM to generate you new project. Of course, it will also run the generated code on your local machine automatically to prove it is working. Once your initial project setup is done, a pull request is provided that can be reviewed by your colleagues. Of course, your colleagues will have access to chat, which is linked in the pull request. Next time one of your colleagues wants to setup another project, they just fork your conversation at some point and continue a new conversation from that point on instead of talking to the LLM from scratch. Since new projects are created frequently in your company, someone is taking your conversation and introduces some parameters to it to use it as a template. Now you have a reusable conversation template that you can simply trigger from your terminal, provide few details (e.g. a project name) and get the code generated.
+
+Of course, not every company creates so many new projects. But this is just one example to show what could be possible, if we were using LLMs more efficiently.
+
+## Designed for your Laptop not the Cloud
+
+The above example gives a rough idea of what you as a software developer should be able to do with Cozy Coder. But to make the tool really good, it obviously must run on your laptop.
+
+Many developers use CLIs and they can easily be used within scripts to automate tasks. As Cozy Coder is designed for developers, it must provide a proper CLI. But also the access to the file system is important to directly modify local files, if you want to.
```

Comment:
```suggestion
Many developers use CLIs and they can easily be used within scripts to automate tasks. As Cozy Coder is designed for developers, it must provide a proper CLI. Also, access to the file system is important to directly modify local files, if you want to.
```

There is no contradiction necessary :) only additions. This makes it feel easier to read.

One last remark about "if you want to". If the point you are trying to convey is really "only if they user _really_ wants to allow the LLM access to the file system" then keep it.

If the point you want to convey is, "when I need the LLM to change things" it is better as ", when you want it to."